test/runtest-all.e2e.js

```diff
@@ -0,0 +1,311 @@
+import { startApp, stopApp, screenshotApp } from './utils'
+import { waitFor } from '@testing-library/dom'
+
+jest.setTimeout(600000)
+
+describe('All network tests run successfully', () => {
+  let app
+
+  beforeAll(async () => {
+    app = await startApp()
+  })
+
+  afterAll(async () => {
+    await stopApp(app)
+  })
+
+  test('Tests start successfully on click of RUN button', async () => {
+    await app.client.$('button[data-testid=button-dashboard-run]').click()
+
+    await waitFor(
```

Comment:
Why is this pattern being used in all the new `.e2e.js` files? It seems like a practice from `testing-library`. Maybe you can explain to me separately how this is supposed to work. I am only wondering if there is a way to avoid writing so much of repetitive code all around.

Reply:
I actually find the `waitFor` api to be the easiest one to implement to assert if a promise is resolved or rejected with a given timeout value.